src/App.tsx

```diff
@@ -1,24 +1,198 @@
+
+  const onSubmitHandler = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (titleTodo) {
+      try {
+        await postTodo(
+          { title: titleTodo, userId: USER_ID, completed: false },
+        );
+
+        getListTodo();
+        setTitleTodo('');
+      } catch (e) {
+        setError('Oops, something were wrong, please try again later');
+      }
+    }
+  };
+
+  const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleTodo(event.target.value);
+  };
+
+  const deleteTodoHandler = (id: number) => {
+    deleteTodo(id).then(() => getListTodo());
+  };
+
+  const changeTodoCompleted = (id: number, completed: boolean) => {
+    patchTodo(id, completed).then(() => getListTodo());
+  };
+
+  const changeTodoTitle = (id: number, title: string) => {
+    patchTodoTitle(id, title).then(() => getListTodo());
+  };
+
+  const clearCompleated = (todos: Todo[]) => {
+    return todos.filter(todo => todo.completed);
+  };
+
+  // const clearActive = (list: Todo[]) => {
+  //   return list.filter(obj => !obj.completed);
+  // };
+
+  const completedTodo = clearCompleated(listTodo);
+  // const activeTodo = clearActive(listTodo);
+
+  function deleteAllTodosHandler(todos: Todo[]) {
+    const promises = todos.map((todo) => deleteTodoHandler(todo.id));
+
+    return Promise.all(promises);
+  }
+
+  function changeAllTodosHandler(todos: Todo[]) {
+    const promises = todos.map((todo) => changeTodoCompleted(todo.id, false));
+
+    return Promise.all(promises);
+  }
+
+  const handleTodoKeyPress = (
+    key: string, id: number, title: string,
+  ) => {
+    if (key === 'Enter') {
+      changeTodoTitle(id, title);
+      setEditTodo(false);
+    }
+
+    if (key === 'Escape') {
+      setEditTodo(false);
+    }
+  };
+
+  const handleTodoBlur = (id: number, title: string) => {
+    changeTodoTitle(id, title);
+    setEditTodo(false);
+  };
+
+  // function changeTodoTitleHandler(id: number, title: string) {
+  //   changeTodoTitle(id, title);
+
+  //   // return Promise.all(promises);
+  // }
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={classNames('todoapp__toggle-all',
+              {
+                active: listTodo.length === completedTodo.length,
+              })}
+            onClick={() => changeAllTodosHandler(completedTodo)}
```

Comment:
No need in extra callback. `changeAllTodosHandler` has todos in upper scope
```suggestion
            onClick={changeAllTodosHandler}
```